pkg/webhook/server.go

```diff
@@ -159,40 +95,75 @@ func NewServer(name string, mgr manager.Manager, options ServerOptions) (*Server
 	as := &Server{
 		Name:          name,
 		sMux:          http.NewServeMux(),
-		registry:      map[string]Webhook{},
+		registry:      map[string]http.Handler{},
 		ServerOptions: options,
 		manager:       mgr,
 	}
 
 	return as, nil
 }
 
+// setDefault does defaulting for the Server.
+func (s *Server) setDefault() {
+	if len(s.Name) == 0 {
+		s.Name = "default-k8s-webhook-server"
+	}
+	if s.registry == nil {
+		s.registry = map[string]http.Handler{}
+	}
+	if s.sMux == nil {
+		s.sMux = http.DefaultServeMux
```

Comment:
> why is the default mux different from the mux used if you call the constructor?

Thanks for catching this.
I agree we should use the same mux. i.e. use `http.NewServeMux()`.
 
> for that matter, why do we have a constructor if we've got the `setDefaults` style?

Because some users may use the public `Server` struct directly, we need to ensure it get sane defaulting and works.